### DISCUSSION TITLE
fastQuery相关问题

### DISCUSSION THREAD

**Discussion by AIsouler** [gkd-kit/gkd#668] [Q&A]
在如下规则中，excludeMatches内的选择器不支持快速查询，matches内选择器支持快速查询，但是fastQuery不能单独针对这两个字段设置，想问一下对于不支持快速查询的选择器设置```fastQuery: true```会不会出现问题?
```
      {
        key: 0,
        fastQuery: true,
        excludeMatches:
          '[text*="搜索" || vid~="(?is).*search.*" || desc*="搜索"][visibleToUser=true]', // 防止在应用的搜索页面误触
        matches: '[text*="跳过"][text.length<10][visibleToUser=true]',
      },
```

**Comment by lisonge** [comment]
<https://gkd.li/selector/optimize#rule-format>

![image](https://github.com/user-attachments/assets/53e5d583-79ca-4147-b585-51fb899d706c)

**Reply by 664873568** [reply]
> https://gkd.li/selector/optimize#rule-format
> 
> ![图像](https://private-user-images.githubusercontent.com/38517192/350482237-53e5d583-79ca-4147-b585-51fb899d706c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3OTAyNzI1MzcsIm5iZiI6MTc5MDI3MjIzNywicGF0aCI6Ii8zODUxNzE5Mi8zNTA0ODIyMzctNTNlNWQ1ODMtNzljYS00MTQ3LWI1ODUtNTFmYjg5OWQ3MDZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA5MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwOTI0VDE3NTAzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUxN2EyMTYwMzhmZmE2NjQ1NTcwOGUwNDM0MDc1ZmU5ODk0ZmU3Mjk2MjNhZDhjZWE0ZDBkZjRmYTZmOWMwMjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.t2x83gC27H9P3U1E8U7vygJRjSMwwflC0815ZeX0qks)

https://gkd.li/guide/optimize#fast-query
fastQuery快速查询逻辑是不是改了呀，规则能快速查询时 才会匹配，如果规则不能快速查询时 就不会匹配 即使规则能查询到节点
